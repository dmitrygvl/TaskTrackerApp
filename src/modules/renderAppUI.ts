export function renderAppUI(rootEl: HTMLElement) {
  rootEl.innerHTML = `
  <main class="main">
        <header>
          <h1>Task Calendar</h1>
        </header>
      <div class="filters">
      <h2>Filters</h2>
        <input type="date" id="filterDate" />
        <select id="filterProgress">
          <option value="new">New</option>
          <option value="in progress">In Progress</option>
          <option value="done">Done</option>
        </select>
        <input
          type="text"
          id="filterTags"
          placeholder="By tags (comma-separated)"
        />
        
        <button id="filterTasksBtn">Filter tasks</button>
      </div>
      <div class="task-form">
        
        <input type="date" id="taskDate" />
        <select id="taskProgress">
          <option value="new">New</option>
          <option value="in progress">In Progress</option>
          <option value="done">Done</option>
        </select>
        <input type="text" id="taskTags" placeholder="Tags (comma-separated)" />
        <input
          id="taskText"
          class="task-text"
          autofocus
          placeholder="Enter your task"
          
          ></input>
        <button type="submit" id="add-update-task-btn">Add/Update Task</button>
      </div>
      <ul class="task-list" id="taskList"></ul>
    </div>
  </main>
  `;
}
