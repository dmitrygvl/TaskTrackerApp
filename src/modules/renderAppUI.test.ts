import { renderAppUI } from './renderAppUI';

describe('renderAppUI', () => {
  it('renders main UI with filters and task form', () => {
    document.body.innerHTML = '<div id="root"></div>';
    const rootEl = document.getElementById('root') as HTMLElement;

    renderAppUI(rootEl);

    expect(rootEl.innerHTML).toMatchSnapshot();
  });

  it('verifies presence of filter elements in the UI', () => {
    document.body.innerHTML = '<div id="root"></div>';
    const rootEl = document.getElementById('root') as HTMLElement;
    renderAppUI(rootEl);

    expect(rootEl.querySelector('#filterDate')).toBeInstanceOf(
      HTMLInputElement,
    );
    expect(rootEl.querySelector('#filterProgress')).toBeInstanceOf(
      HTMLSelectElement,
    );
    expect(rootEl.querySelector('#filterTags')).toBeInstanceOf(
      HTMLInputElement,
    );
    expect(rootEl.querySelector('#filterTasksBtn')).toBeInstanceOf(
      HTMLButtonElement,
    );
  });

  it('verifies presence of task form elements in the UI', () => {
    document.body.innerHTML = '<div id="root"></div>';
    const rootEl = document.getElementById('root') as HTMLElement;
    renderAppUI(rootEl);

    expect(rootEl.querySelector('#taskDate')).toBeInstanceOf(HTMLInputElement);
    expect(rootEl.querySelector('#taskProgress')).toBeInstanceOf(
      HTMLSelectElement,
    );
    expect(rootEl.querySelector('#taskTags')).toBeInstanceOf(HTMLInputElement);
    expect(rootEl.querySelector('#taskText')).toBeInstanceOf(HTMLInputElement);
    expect(rootEl.querySelector('#add-update-task-btn')).toBeInstanceOf(
      HTMLButtonElement,
    );
  });

  it('verifies the presence of task list in the UI', () => {
    document.body.innerHTML = '<div id="root"></div>';
    const rootEl = document.getElementById('root') as HTMLElement;
    renderAppUI(rootEl);

    expect(rootEl.querySelector('#taskList')).toBeInstanceOf(HTMLUListElement);
  });
});
