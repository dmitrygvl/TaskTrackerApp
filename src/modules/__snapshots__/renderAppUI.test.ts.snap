// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderAppUI renders main UI with filters and task form 1`] = `
"
  <main class="main">
        <header>
          <h1>Task Calendar</h1>
        </header>
      <div class="filters">
      <h2>Filters</h2>
        <input type="date" id="filterDate">
        <select id="filterProgress">
          <option value="new">New</option>
          <option value="in progress">In Progress</option>
          <option value="done">Done</option>
        </select>
        <input type="text" id="filterTags" placeholder="By tags">
        
        <button class="filters__btn" id="filterTasksBtn">Filter tasks</button>
      </div>
      <div class="task-form">
        
        <input type="date" id="taskDate">
        <select id="taskProgress">
          <option value="new">New</option>
          <option value="in progress">In Progress</option>
          <option value="done">Done</option>
        </select>
        <input type="text" id="taskTags" placeholder="Tags">
        <input id="taskText" class="task-text" autofocus="" placeholder="Enter your task">
        <button type="submit" id="add-update-task-btn">Add/Update Task</button>
      </div>
      <ul class="task-list" id="taskList"></ul>
    
  </main>
  "
`;
